crates/ruma-identifiers/src/mxc_uri.rs

```diff
@@ -38,9 +43,14 @@ impl MxcUri {
         })
     }
 
+    /// If an error occurred during parsing, retrieve it via here.
+    pub fn inner_err(&self) -> Option<MxcUriError> {
+        self.slash_idx.err()
+    }
+
     /// Returns if this is a spec-compliant MXC URI.
     pub fn is_valid(&self) -> bool {
-        self.slash_idx.is_some()
+        self.slash_idx.is_ok()
     }
```

Comment:
Instead of these two fns, I think we should just have one `.validate()` that's the same as `.parts().map(|_| ())`.

Reply:
See above comment, i dont know if this is the right way, it does another computation just to throw away the result

Reply:
I think if one wants to parse the URI only for the purpose of knowing whether it's valid, then this seems entirely reasonable to me. Of course it would be stupid to use `.validate` right before `.parts`, but I think it should be clear enough from `.parts`' signature that it does the same checks on its own.

Reply:
Eh, alright, both functions still "gate" their values behind `Result`, so i'll add that 👍